pkg/functions/riff/driver.go

```diff
@@ -116,14 +140,15 @@ func (d *riffDriver) GetRunnable(e *functions.FunctionExecution) functions.Runna
 
 		resBytes, err := d.requester.Request(topic, e.RunID, bytesIn)
 		if err != nil {
-			return nil, errors.Wrapf(err, "riff: error invoking function: '%s', runID: '%s'", e.FunctionID, e.RunID)
+			return nil, &systemError{errors.Wrapf(err, "riff: error invoking function: '%s', runID: '%s'", e.FunctionID, e.RunID)}
 		}
 
 		var out functions.Message
 		if err := json.Unmarshal(resBytes, &out); err != nil {
-			return nil, errors.Errorf("cannot JSON-parse result from riff: %s %s", err, string(resBytes))
+			return nil, &outputError{errors.Errorf("cannot JSON-parse result from riff: %s %s", err, string(resBytes))}
```

Comment:
This one is a system error too: it's the function runtime responsible for JSON-encoding the response, not the function.